packages/gatsby/src/schema/types/node-interface.ts

```diff
@@ -64,10 +64,6 @@ const getOrCreateNodeInterface = <TSource, TArgs>(
       },
       internal: internalTC.getTypeNonNull(),
     })
-    // TODO: In Gatsby v2, the NodeInput.id field is of type String, not ID.
-    // Remove this workaround for v3.
-    const nodeInputTC = tc.getInputTypeComposer()
-    nodeInputTC.extendField(`id`, { type: `String` })
```

Comment:
Will this force users to change their query variables:
```diff
-query(id: String!)
+query(id: ID!)
```
?

Reply:
Yes, you are right. That's probably why those e2e tests are also failing. This makes me even more hesitant about actually doing this.